packages/js/components/src/select-control/select-control.tsx

```diff
@@ -0,0 +1,162 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { createElement } from 'react';
+import { useCombobox, useMultipleSelection } from 'downshift';
+import { useState, Fragment } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { ChildrenType, ItemType } from './types';
+import { SelectedItems } from './selected-items';
+import { ComboBox } from './combo-box';
+import { Menu } from './menu';
+import { MenuItem } from './menu-item';
+import {
+	itemToString as defaultItemToString,
+	getFilteredItems as defaultGetFilteredItems,
+} from './utils';
+
+type SelectControlProps = {
+	children?: ChildrenType;
+	items: ItemType[];
+	label: string;
+	initialSelectedItems?: ItemType[];
+	itemToString?: ( item: ItemType | null ) => string;
+	getFilteredItems?: (
+		allItems: ItemType[],
+		inputValue: string,
+		selectedItems: ItemType[]
+	) => ItemType[];
+	multiple?: boolean;
+	onInputChange?: ( value: string | undefined ) => void;
+	onRemove?: ( item: ItemType ) => void;
+	onSelect?: ( selected: ItemType ) => void;
+	placeholder?: string;
+	selected: ItemType | ItemType[] | null;
+};
+
+export const SelectControl = ( {
+	children = ( {
+		items,
+		highlightedIndex,
+		getItemProps,
+		getMenuProps,
+		isOpen,
+	} ) => {
+		return (
+			<Menu getMenuProps={ getMenuProps } isOpen={ isOpen }>
+				{ items.map( ( item, index: number ) => (
+					<MenuItem
+						key={ `${ item.value }${ index }` }
+						index={ index }
+						isActive={ highlightedIndex === index }
+						item={ item }
+						getItemProps={ getItemProps }
+					>
+						{ item.label }
+					</MenuItem>
+				) ) }
+			</Menu>
+		);
+	},
+	multiple = false,
+	items,
+	label,
+	itemToString = defaultItemToString,
+	getFilteredItems = defaultGetFilteredItems,
+	onInputChange = () => null,
+	onRemove = () => null,
+	onSelect = () => null,
+	placeholder,
+	selected,
```

Comment:
Could we also have an option where we can open on focus?

Reply:
Are you looking for an `onFocus` prop or a way to open the menu when no search query has occurred?  In both cases, I would save for a follow-up.

Reply:
Depends I guess, if I have the ability to open the menu up outside of the `SelectControl` context then a `onFocus` callback would be great.
If the `isOpen` handling will stay within `SelectControl`, an option for `openOnFocus` would be helpful.

Reply:
If you override the render with children, you will get access to `isOpen`, but I'm still considering something like an `openOnFocus`.  I was also thinking of introducing this as both an `onFocus` callback and `openMenu`, but still a little on the fence with this.

I'm hoping to see how we use this component in action before settling on one of these paths.